File: 백준/Silver/1475. 방 번호/방 번호.java

import java.lang.reflect.Array;
import java.util.Arrays;
import java.util.Scanner;

public class Main{
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        String num = sc.next();
        int cnt = 0;

        int[] arr = new int[10];

        for(int i=0; i<num.length(); i++){
            arr[num.charAt(i)-'0']++;
        }


        for(int i=0; i<10; i++){
            if(arr[6]!=0 || arr[9]!=0) {
                int mid = (arr[6]+ arr[9]+1)/2;
                arr[6] = mid;
                arr[9] = mid;
            }
        }


        System.out.println(Arrays.stream(arr).max().getAsInt());
    }
}

File: 백준/Silver/1475. 방 번호/방 번호.py
n = input()

visited = [0]*10

for i in n:
    num = int(i)
    if num ==6 or num ==9:
        if visited[6]< visited[9]:
            num = 6
        else:
            num = 9

    visited[num] += 1
print(max(visited))